core/src/components/segment/segment.tsx

```diff
@@ -373,10 +382,18 @@ export class Segment implements ComponentInterface {
 
   private onClick = (ev: Event) => {
     const current = ev.target as HTMLIonSegmentButtonElement;
+    /**
+     * If current element is ion-segment then that means
+     * user tried to swipe to other ion-segment-button,
+     * and we should not update the checked button.
+     */
+    if (current.tagName === 'ION-SEGMENT') {
```

Comment:
Can you clarify why this is needed?

Reply:
What point should I clarify? Without these lines, the animation doesn't work if the user tried to swipe to then click another button.

Reply:
I am mostly curious what issue this fixed by adding that line

Reply:
I see🙂 The change or comment itself is ok?

Reply:
Yeah change + comment looks good, thanks!